src/test/java/com/karankumar/bookproject/ui/book/BookFormTest.java

```diff
@@ -479,14 +517,169 @@ void shouldDisplaySeriesPosition_withSeriesPositionPopulated_whenBookHasSeriesPo
+
+        // then
+        Assertions.assertEquals(1, bookService.count());
+        Book bookInDatabase = bookService.findAll().get(0);
+        assertBookInSpecificShelf(TO_READ, bookInDatabase);
+    }
+
+    /**
+     * Tests whether a book is updated in the database when the save-event is called
+     */
+    @Test
+    void shouldUpdateValuesForBook() {
+        // given
+        bookForm = createBookForm(TO_READ, false);
+        bookForm.addListener(BookForm.SaveEvent.class, event -> bookService.save(event.getBook()));
+        bookForm.saveButton.click();
+
+        Book savedBook = bookService.findAll().get(0);
+        Assertions.assertEquals(bookTitle, savedBook.getTitle());
+        Assertions.assertEquals(TO_READ, savedBook.getPredefinedShelf().getPredefinedShelfName());
+        Assertions.assertEquals(firstName, savedBook.getAuthor().getFirstName());
+        Assertions.assertEquals(lastName, savedBook.getAuthor().getLastName());
+        Assertions.assertEquals(genre, savedBook.getGenre());
+        Assertions.assertNull(savedBook.getDateStartedReading());
+        Assertions.assertNull(savedBook.getDateFinishedReading());
+
+        populateBookForm(READ, false, savedBook);
+        bookForm.bookTitle.setValue("IT");
+        bookForm.authorFirstName.setValue("Stephen");
+        bookForm.authorLastName.setValue("King");
+        bookForm.bookGenre.setValue(Genre.HORROR);
+
+        // when
+        bookForm.saveButton.click();
+
+        // then
+        Assertions.assertEquals(1, bookService.count());
+        Book updatedBook = bookService.findAll().get(0);
+        Assertions.assertEquals("IT", updatedBook.getTitle());
+        Assertions.assertEquals(READ, updatedBook.getPredefinedShelf().getPredefinedShelfName());
+        Assertions.assertEquals("Stephen", updatedBook.getAuthor().getFirstName());
+        Assertions.assertEquals("King", updatedBook.getAuthor().getLastName());
+        Assertions.assertEquals(Genre.HORROR, updatedBook.getGenre());
+        Assertions.assertEquals(dateStarted, updatedBook.getDateStartedReading());
+        Assertions.assertEquals(dateFinished, updatedBook.getDateFinishedReading());
+    }
+
+
+    private static Stream<Arguments> shelfCombinations() {
+        return Stream.of(
+                Arguments.of(TO_READ, READING),
+                Arguments.of(TO_READ, READ),
+                Arguments.of(TO_READ, DID_NOT_FINISH),
+                Arguments.of(READING, DID_NOT_FINISH),
+                Arguments.of(READING, READ)
+                // TODO: these cases are not passing atm because of issue #271
+                // Arguments.of(READING, TO_READ),
+                // Arguments.of(READ, TO_READ),
+                // Arguments.of(READ, DID_NOT_FINISH),
+                // Arguments.of(READ, READING),
+                // Arguments.of(DID_NOT_FINISH, TO_READ),
+                // Arguments.of(DID_NOT_FINISH, READING),
+                // Arguments.of(DID_NOT_FINISH, READ)
```

Comment:
Just a question, is it better to merge with commented code or raise another PR when #271 is fixed?

Reply:
My thought was, that by adding them, the person working on #271 sees these tests and can check their solution with them.

Reply:
Tom and I have [discussed this](https://github.com/knjk04/book-project/issues/90#issuecomment-679132648) in #90. The rationale for keeping commented out code was that we don't want to forego the tests. We need the tests, we just need to get them to work. Todos are fine as long as there's a coresponding issue.